### ISSUE TITLE
Migrate SB and Unified Build Installer Pipelines to SDK

### ISSUE THREAD

**Issue by ellahathaway** [dotnet/source-build#4357]
As part of the work to [merge Installer to SDK](https://github.com/dotnet/sdk/pull/38804), the SB pipelines will need to be updated to be run out of SDK. This work should be completed concurrently with the work to update the synchronization (@ViktorHofer).

This pipeline work is broken into a few steps:
- [x] Copy over installer/eng/pipelines into sdk/eng/pipelines: https://github.com/dotnet/sdk/pull/40420
- [x] Modify ci.yml to use sdk/eng/pipelines templates instead of installer templates: https://github.com/dotnet/sdk/pull/40451
- [x] Create duplicate of the [internal installer-source-build pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=1232&_a=summary) for sdk: [NEW PIPELINE](https://dev.azure.com/dnceng/internal/_build?definitionId=1378&_a=summary) (internal Microsoft link)
- [x] Create duplicates of the PR validation pipelines
  - [x] The rules will need to be changed to match those of installer in terms of using the new SDK pipelines https://github.com/dotnet/installer/settings/branch_protection_rules/1970909 - Waiting on green SB and UB builds out of SDK.
- [x] Create a branch of the SB release pipeline that releases from the SDK for 9.0 instead of installer. Once this is working and we've deprecated installer in main, the actual SB release pipeline should be updated to include the changes in this branch.
  - [DRAFT PR](https://dev.azure.com/dnceng/internal/_git/dotnet-release/pullrequest/39308) (internal Microsoft link).
  - Waiting to publish until Staging changes go in, but otherwise changes in the draft PR are ready.
- [x] Update the SB monitor dashboard to include main-sdk branch

**Comment by ellahathaway** [comment]
Current progress:
- dotnet-source-build-sdk pipeline configured, based on my internal Azdo dev branch
- dotnet-source-build-lite-sdk pipeline configured, based on my internal Azdo dev branch

**Comment by ViktorHofer** [comment]
AFAIK the only thing remaining is

> Create a shadow of the SB release pipeline that releases from the SDK instead of installer. Once this is working and we've deprecated main in installer, the actual SB release pipeline should be updated to match this "shadow".

The sync now works and we will soon hopefully have green SB and unified-build PR builds out of sdk.

**Comment by premun** [comment]
@ellahathaway don't we need to keep the installer-based release pipeline around? For 8.0 releases? Possibly even for 7.0 and 6.0?

**Comment by ellahathaway** [comment]
> don't we need to keep the installer-based release pipeline around? For 8.0 releases? Possibly even for 7.0 and 6.0?

Absolutely! The current infra of the source-build release pipeline allows for a simple change to be made in order to use of the sdk-based VMR instead of the installer-based VMR. This change is localized to 9.0, meaning that the pipeline will still support installer-based releases for <= 8.0.

**Comment by ellahathaway** [comment]
Last remaining step is blocked on the staging pipeline work.

**Comment by ellahathaway** [comment]
The PR has gone in, so I'm closing this issue.
